db/db_creator.py

```diff
@@ -1,70 +1,73 @@
+# This is needed for type hinting with fluent interfaces
+from __future__ import annotations
+
 # standard libraries
 import logging, os
-from typing import Dict, Sequence
+from typing import Dict, List
 from urllib.parse import quote_plus
 
 # third-party libraries
-from sqlalchemy.engine import create_engine
+from sqlalchemy.engine import create_engine, Engine
 from yoyo import get_backend, read_migrations
 
+
+# Initialize settings and global variables
+
 logger = logging.getLogger(__name__)
 
-ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
-MIGRATIONS_PATH = os.path.join(ROOT_DIR, 'migrations')
+PARENT_PATH = os.path.dirname(os.path.abspath(__file__))
+MIGRATIONS_PATH = os.path.join(PARENT_PATH, 'migrations')
+
+# The metadata tables are append tables by default.
+DEFAULT_APPEND_TABLE_NAMES = ['job_run', 'data_source_status']
 
 
 class DBCreator:
 
     def __init__(
         self,
         db_params: Dict[str, str],
-        append_table_names: Sequence[str] = []
+        append_table_names: List[str] = []
     ) -> None:
 
-        self.db_name = db_params['dbname']
-        self.conn = None
-        self.conn_str = (
+        self.db_name: str = db_params['dbname']
+        self.conn_str: str = (
             'mysql+mysqldb' +
             f"://{db_params['user']}" +
             f":{quote_plus(db_params['password'])}" +
             f"@{db_params['host']}" +
             f":{db_params['port']}" +
             f"/{db_params['dbname']}?charset=utf8&ssl=true"
         )
-        self.engine = create_engine(self.conn_str)
-        self.append_table_names = append_table_names
-
-    def set_up(self) -> None:
-        logger.debug('set_up')
-        self.conn = self.engine.connect()
```

Comment:
I suspect removing `connect()` from here will break the MiVideo code.

Reply:
Nope, it will not. You're connecting via `self.engine`, which is how I'm recommending people do this in the future.

Reply:
Note that I did change a couple lines of `mivideo/extract.py`.

Reply:
Yes.  I just didn't try running a copy of this myself to see whether this change would cause a problem.